### DISCUSSION TITLE
ponder while oponent is thinking

### DISCUSSION THREAD

**Discussion by danielgimeno** [official-stockfish/Stockfish#3622] [Q&A]
Hi there, I have developed a client gui, for personal use, Peer to Peer, Peer to Sf and Sf toSf. Everything seems to work perfectly, but one thing, the ponder mode True is not using oponent time to think. A Simple htop Linux command reveals that  while Sf is think ing the 8 (4x2) cores of my machine are at 100% CPU usage, but as soon as the bestmove is delivered, all cores goes again to 0%.
I use a "setoption name ponder value True" command, but it seems to be not correct. Which is the proper way to use all time thinking?
Thank you in advance for you time,  and for offering this amazing engine.
Kind regards, Daniel.

**Comment by niklasf** [comment]
The [option itself](https://backscattering.de/chess/uci/#engine-option-ponder) is just used to let the engine know that it will get opportunities to ponder. For example the engine might change its time management strategy based on that.

To actually ponder:

1. The last "foreground" search may have returned a hint for a move to ponder on ([`bestmove ... ponder ...`](https://backscattering.de/chess/uci/#engine-bestmove)).
2. If so, when the opponent starts thinking, set up the position after the ponder move.
3. Actually start the ponder search with [`go ponder`](https://backscattering.de/chess/uci/#gui-go-ponder). Can also specify the usual limits, but they will not yet go into effect.
4. Once the opponent played the expected ponder move, send [`ponderhit`](https://backscattering.de/chess/uci/#gui-ponderhit) to convert the ponder search into a foregound search (with the prepared limits). If they played a different move, just stop, set up the position and start a new normal foreground search.
